feat: add crd validation

thxCode opened the pull request:
**Fixes**:

https://github.com/rancher/harvester/issues/56

**Solution**:

- support to generate CRD validation spec.
- separate the CRD initialization from global setup.
- adjust the CRD initialization logic so that it doesn't overwrite the existing CRD.

LLParse commented:
Closing due to age. We've since adopted build args and images are `FROM scratch`.

Commit:
feat: add crd validation

- support to generate CRD validation spec.
- separate the CRD initialization from global setup.
- adjust the CRD initialization logic so that it doesn't overwrite the
  existing CRD.

orangedeng reviewed (approved):
LGTM

guangbochen reviewed (approved):
LGTM
